CCScriptWriter/CCScriptWriter.py

```diff
@@ -478,16 +379,60 @@ def loadDialogue(self, loadCoilSnake=False):
                                    FormatHex(self.data[a + 7]),
                                    FormatHex(self.data[a + 8])))
                 t = 1
+            else:
+                assert False
             m = self.dataFiles[address]
             h = hex(address)
             self.asmPointers[a] = ["{}.l_{}".format(m, h), t]
 
+    def loadCoilSnakeDialogue(self):
+        "Load pointers from the CoilSnake project."
+        o = os.path.join(self.outputDirectory, os.path.pardir)
+        project = os.path.join(o, "Project.snake")
+        try:
+            with open(project, "r", encoding="utf8"):
+                pass
+        except IOError:
+            print("Failed to open \"{}\". Invalid CoilSnake project. "
+                    "Aborting.".format(project))
+            sys.exit(1)
+        # pylint: disable=too-many-nested-blocks
+        for fileName in COILSNAKE_FILES:
+            with open(os.path.join(o, fileName), "r", encoding="utf8") as csFile:
+                yamlData = yaml.load(csFile, Loader=yaml.CSafeLoader)
+            if fileName != "map_doors.yml":
+                for v in yamlData.values():
+                    for p in COILSNAKE_POINTERS:
+                        if p in v:
+                            try:
+                                pointer = int(v[p][1:], 16)
+                                if pointer > 0xc00000 \
+                                    and pointer not in self.dialogue:
+                                    self.pointers.append(pointer)
+                            except ValueError:
+                                self.pointers.append(int(v[p][12:], 16))
```

Comment:
```suggestion
                            try:
                                # Assume pointer is a string beginning with '$'
                                pointer = int(v[p][1:], 16)
                                if pointer > 0xc00000 \
                                    and pointer not in self.dialogue:
                                    self.pointers.append(pointer)
                            except ValueError:
                                # We may have previously begun decompiling the script into this project.
                                # Instead assume the label begins with `data_##.l_0x`
                                self.pointers.append(int(v[p][12:], 16))
```
The intention of this code isn't super obvious when reading through it. (Alternatively, these might be good spots to insert another regex verification and give better error messages complaining that it doesn't look like a vanilla label?)

Ditto for the instance below this

Reply:
Refactored the string -> int processing into its own function which uses a regex - I think it's inherently more readable. Let me know what you think!